Benchmarks: memory footprint test

chen-png opened the pull request:
samples/basic/minial: add some instances for mec15xxevb_assy6853 board.

add some test instances to test memory footprint change
on mec15xxevb_assy6853 board with different configuration,
such as enable multithreading, network and bluetooth.


Signed-off-by: peng1 chen <peng1.chen@intel.com>

daee-demant committed:
bluetooth: host: Do not send unwanted SC indicate

Fixes #23485

When we create a GATT table dynamically, we also create a hash
identifying this table. This hash can be stored in persistent memory and
we can thus determine after recreating the GATT table whether the
services have changed or not from before the reboot.

When these hashes are identical, it implies that the table has not
changed, wherefore a service changed indication should not be sent to
any bonded clients. The method for achieving this was to remove the
gatt_sc.work entry from the work queue. This work queue entry was to
send an indication to the clients when the table had been allocated.
If the final entry then caused the hashes to match, the indication
would be cancelled.

On unit testing this behaviour in simulation and in practice, we found
that the indication was sent nonetheless, and the issue was located to
be tied to the SERVICE_RANGE_CHANGED flag which is set when the services
are changed and is cleared when the indications are being sent out.

It was the job of the work queue entry to clear this flag, and as the
entry was never serviced, the flag was never cleared, and when
sc_commit() is called at the end of the process, it believes that there
is a new service change pending and therefore starts the job over, thus
creating a redundant indication to the clients.

This commit fixes the issue by clearing the flag when the work entry
is removed due to a hash match. This has been unittested in a live
environment, in a simulation environment, and sanitycheck has been run
on it.

Signed-off-by: Dan Erichsen <daee@demant.com>

wopu-ot committed:
Bluetooth: controller: split: Fix response to unexpected LL_FEATURE_RSP

Fix response to unexpected LL_FEATURE_RSP for the case that
BT_CTLR_SLAVE_FEAT_REQ is disabled. Fixes LL/PAC/SLA/BV-01 for such a
configuration.

Signed-off-by: Wolfgang Puffitsch <wopu@demant.com>

Commit:
Merge branch 'master' of https://github.com/zephyrproject-rtos/zephyr into memory_footprint_test

carlescufi reviewed (changes requested):
We already have `samples/basic/minimal`. This seems to duplicate that to some extent, is it necessary?

nashif reviewed (changes requested):
Not sure what is the intention of this PR and we have something already as @carlescufi  pointed out.

chen-png commented:
i want to test the memory footprint on mec15xxevb_assy6853 board, but if i used the samples/basic/minimal, it will meet many build error, because the configuration file is not suitable for mec board.

also using the samples/basic/minimal, i can build a smaller size binary file, but i am not sure whether it could be ran on real hardware boards, because i could not get anything from the console, so i added a print line "hello world" in test case.

carlescufi commented:
> i want to test the memory footprint on mec15xxevb_assy6853 board, but if i used the samples/basic/minimal, it will meet many build error, because the configuration file is not suitable for mec board.

But you can expand the set of configuration files in that sample to fit your board perhaps instead? I am sure there's a way to do that using the current project.
 
> also using the samples/basic/minimal, i can build a smaller size binary file, but i am not sure whether it could be ran on real hardware boards, because i could not get anything from the console, so i added a print line "hello world" in test case.

Right so how about adding a new configuration in the very same minimal project that enables the console and conditionally prints out a message?

```
#fidef CONFIG_UART_CONSOLE
printk("Test OK")
#endif
```

chen-png commented:
@carlescufi  thanks, i will make change based on samples/basic/minimal test.

Commit:
samples/basic/minial: add some instances for mec15xxevb_assy6853 board.

add some test instances to test memory footprint change
on mec15xxevb_assy6853 board with different configuration,
such as enable multithreading, network and bluetooth.

Signed-off-by: peng1 chen <peng1.chen@intel.com>

LeiW000 commented on samples/basic/minimal/sample.yaml:
```diff
@@ -42,3 +42,19 @@ tests:
     build_only: true
     extra_args: CONF_FILE='common.conf;no-mt.conf;no-timers.conf;x86.conf'
     platform_whitelist: qemu_x86
+  sample.minimal.mec15xxevb_assy6853:
```
Please don't put platform name in the test case name.

chen-png commented:
@carlescufi  could you take a look of this?

carlescufi commented:
I am really not sure that adding bluetooth and networking to a "minimal" sample is a good idea. Instead why not add a specific `.conf` file to existing bluetooth and networking samples for your platform, and then use those samples to track the footprint?

carlescufi commented on samples/basic/minimal/mec.conf:
```diff
@@ -0,0 +1,41 @@
+# Drivers and peripherals
```
Do you need this file? isn't it equivalent to a combination of the arm + no-mt configurations?

chen-png replied:
if i used the common.conf , for mec15xxevb_assy6953 board, it will report a build failure.
for example, if i disabled CONFIG_PINMUX as common.conf,  it failed with 
 "error: 'porta' undeclared (first use in this function)
  152 |  pinmux_pin_set(porta, MCHP_GPIO_000, MCHP_GPIO_CTRL_MUX_F0);"
so i creaed this file for mec board.

chen-png commented:
> I am really not sure that adding bluetooth and networking to a "minimal" sample is a good idea. Instead why not add a specific `.conf` file to existing bluetooth and networking samples for your platform, and then use those samples to track the footprint?

you mean i should just add my board to run this samples/basic/minimal test case, and for network and bluetooth, i should use other test case, right?

nashif commented:
@chen-png  Please do not add specific board configurations and testcases to this, if you need to track footprint for some specific hardware, this should be done out of the mainline tree, we are not tracking data for specific boards here. This PR will not be merged as it is now.

chen-png commented:
i got it, i will try to add my board just using existed config,  rather than specifying a special config.
